src/v/datalake/translation/state_machine.cc

```diff
@@ -210,7 +210,7 @@ ss::future<> translation_stm::apply_raft_snapshot(const iobuf&) {
     // with the snapshot.
     vlog(_log.debug, "Applying raft snapshot, resetting state");
     _highest_translated_offset = kafka::offset{};
-    _last_translated_timestamp = model::timestamp::min();
+    _last_translated_timestamp = std::nullopt;
```

Comment:
There's also `timestamp::min()` in `do_apply()` that should probably be removed?

Reply:
that one is intentional - if `value.last_translated_timestamp.has_value()`, then its value must be `>= timestamp::min()`, so we can just resolve the cached copy to `timestamp::min()` to make it work w/ `std::max`. 

though i just remembered that `std::optional` supports comparison operators, so I believe this would have the same effect:

```c++
_last_translated_timestamp = std::max(_last_translated_timestamp, value.last_translated_timestamp);
```

in the sense of always blowing away the cached copy if it's nullopt and comparing the two otherwise.